src/coreclr/inc/corinfo.h

```diff
@@ -3014,15 +3014,19 @@ class ICorStaticInfo
             size_t*               pRequiredBufferSize = nullptr
             ) = 0;
 
-    // Return method name as in metadata, or nullptr if there is none,
-    // and optionally return the class, enclosing class, and namespace names
-    // as in metadata.
+    // Return method name as in metadata, or nullptr if there is none, and
+    // optionally return the class, enclosing classes, and namespace name as
+    // in metadata. Enclosing classes are returned from inner-most enclosed class
+    // to outer-most, with nullptr in the array indicating that no more
```

Comment:
Is the caller expected to always check that the returned enclosingClassNames is terminated by null?

Otherwise, there does not seem to be any other way to tell whether the enclosingClassNames buffer was sufficient or whether it may contain partial data.

Reply:
For the JIT we are only using the function to check for known methods, so we don't end up needing to do that. We just know that a buffer of 2 is sufficient to get the enclosing classes and namespace back for all intrinsic functions we want to recognize.
If we wanted to use this function to reconstruct the full name of an arbitrary function then I think we would need to do something like what you suggest with the current way the API is shaped.

Reply:
I am worried about false positives. What happens if somebody introduces triple-nested method that happens to match double nested method that you recognize?

Reply:
The function returns the namespace for the outer most class returned. For a triple nested class the two inner classes do not have a namespace, thus on the JIT side we would not get any namespace back to recognize a class in.

Reply:
ok